src/safir/dependencies/arq.py

```diff
@@ -0,0 +1,592 @@
+        _queue_name : `str`
+            Name of the queue.
+        **kwargs
+            Keyword arguments passed to the task function.
+
+        Returns
+        -------
+        JobMetadata
+            Metadata about the queued job.
+
+        Raises
+        ------
+        JobNotQueued
+            Raised if the job is not successfully added to the queue.
+        """
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    async def get_job_metadata(
+        self, job_id: str, queue_name: Optional[str] = None
+    ) -> JobMetadata:
+        """Get metadata about a `~arq.jobs.Job`.
+
+        Parameters
+        ----------
+        job_id : `str`
+            The job's identifier. This is the same as the `JobMetadata.id`
+            attribute, provided when initially adding a job to the queue.
+        queue_name : `str`, optional
+            Name of the queue.
+
+        Returns
+        -------
+        `JobMetadata`
+            Metadata about the queued job.
+
+        Raises
+        ------
+        JobNotFound
+            Raised if the job is not found in the queue.
+        """
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    async def get_job_result(
+        self, job_id: str, queue_name: Optional[str] = None
+    ) -> JobResult:
+        """The job result, if available.
+
+        Parameters
+        ----------
+        job_id : `str`
+            The job's identifier. This is the same as the `JobMetadata.id`
+            attribute, provided when initially adding a job to the queue.
+        queue_name : `str`, optional
+            Name of the queue.
+
+        Returns
+        -------
+        `JobResult`
+            The job's result, along with metadata about the queued job.
+
+        Raises
+        ------
+        JobNotFound
+            Raised if the job is not found in the queue.
+        JobResultUnavailable
+            Raised if the job's result is unavailable for any reason.
+        """
+        raise NotImplementedError
+
+
+class RedisArqQueue(ArqQueue):
+    """A distributed queue, based on arq and Redis."""
+
+    def __init__(
+        self,
+        pool: ArqRedis,
+        *,
+        default_queue_name: str = arq_default_queue_name,
+    ) -> None:
+        super().__init__(default_queue_name=default_queue_name)
+        self._pool = pool
+
+    @classmethod
+    async def initialize(
+        cls,
+        redis_settings: RedisSettings,
+        *,
+        default_queue_name: str = arq_default_queue_name,
+    ) -> RedisArqQueue:
+        """Initialize a RedisArqQueue from Redis settings."""
+        pool = await create_pool(
+            redis_settings, default_queue_name=default_queue_name
+        )
+        return cls(pool)
+
+    @property
+    def default_queue_name(self) -> str:
+        """Name of the default queue, if the ``_queue_name`` parameter is
+        no set in method calls.
+        """
+        return self._default_queue_name
```

Comment:
This doesn't look different from the parent class implementation. I think you can delete it unless I'm missing something.